tests/org/mozilla/io/TestLocalMsgProtocol.java

```diff
@@ -353,26 +309,13 @@ public void test(TestHarness th) {
             // Scenario 1
             Thread serverCreateThread = new ThreadServerCreate();
             serverCreateThread.start();
-            synchronized (openLock) {
-                while (!serverCreated) {
-                    openLock.wait();
-                }
-            }
+            serverCreateThread.join();
             Thread serverAcceptAndOpenThread = new ThreadServerAcceptAndOpen();
             serverAcceptAndOpenThread.start();
-            synchronized (openLock) {
-                while (!serverAcceptAndOpenCalled) {
-                    openLock.wait();
-                }
-            }
-            Thread clientThread = new ThreadClient();
+            Thread clientThread = new ThreadClientCreate();
             clientThread.start();
-            serverCreateThread.join();
             serverAcceptAndOpenThread.join();
```

Comment:
Yes, in general it could, but it doesn't with the way the scheduler works now. It's definitely a weak assumption, so it's great you're fixing it.